supervision/detection/core.py

```diff
@@ -483,11 +484,40 @@ class names. If provided, the resulting Detections object will contain
             Class names values can be accessed using `detections["class_name"]`.
         """  # noqa: E501 // docs
 
-        class_ids = transformers_results["labels"].cpu().detach().numpy().astype(int)
         data = {}
```

Comment:
In `from_transformers`, there are four places where we do:

```python
if id2label is not None:
    class_names = np.array([id2label[class_id] for class_id in class_ids])
    data[CLASS_NAME_DATA_FIELD] = class_names
```

I'd wrap it in a local helper function. It can be defined inside `from_transformers`.

```python
def get_data(class_ids: np.ndarray, id2label: Optional[Dict[int, str]]) -> dict:
    data = {}
    if id2label is not None:
        class_names = np.array([id2label[class_id] for class_id in class_ids])
        data[CLASS_NAME_DATA_FIELD] = class_names
    return data
```